test/e2e/set_delete_test.go

```diff
@@ -0,0 +1,249 @@
+
+var podCount int
+
+func testHeaders(t *testing.T, image string, route *routev1.Route, address string, headers []string, expectedResponse string, expectedMatches int) {
+	t.Helper()
+
+	kubeConfig, err := config.GetConfig()
+	if err != nil {
+		t.Fatalf("failed to get kube config: %v", err)
+	}
+	client, err := kubernetes.NewForConfig(kubeConfig)
+	if err != nil {
+		t.Fatalf("failed to create kube client: %v", err)
+	}
+
+	var extraCurlArgs []string
+	for _, header := range headers {
+		extraCurlArgs = append(extraCurlArgs, "-H", header)
+	}
+	extraCurlArgs = append(extraCurlArgs, "-v", "--resolve", route.Spec.Host+":80:"+address)
+	podCount++
+	name := fmt.Sprintf("%s%d", route.Name, podCount)
+	clientPod := buildCurlPod(name, route.Namespace, image, route.Spec.Host, address, extraCurlArgs...)
+	if err := kclient.Create(context.TODO(), clientPod); err != nil {
+		t.Fatalf("failed to create pod %s/%s: %v", clientPod.Namespace, clientPod.Name, err)
+	}
+	defer func() {
+		if err := kclient.Delete(context.TODO(), clientPod); err != nil {
+			if !errors.IsNotFound(err) {
+				t.Fatalf("failed to delete pod %s/%s: %v", clientPod.Namespace, clientPod.Name, err)
+			}
+		}
+	}()
+	expectedResponse = strings.ToLower(expectedResponse)
+	err = wait.PollImmediate(1*time.Second, 4*time.Minute, func() (bool, error) {
+		readCloser, err := client.CoreV1().Pods(clientPod.Namespace).GetLogs(clientPod.Name, &corev1.PodLogOptions{
+			Container: "curl",
+			Follow:    false,
+		}).Stream(context.TODO())
+		if err != nil {
+			t.Logf("failed to read output from pod %s: %v", clientPod.Name, err)
+			return false, nil
+		}
+		scanner := bufio.NewScanner(readCloser)
+		defer func() {
+			if err := readCloser.Close(); err != nil {
+				t.Errorf("failed to close reader for pod %s: %v", clientPod.Name, err)
+			}
+		}()
+		for scanner.Scan() {
+			line := scanner.Text()
+			if strings.Contains(strings.ToLower(line), expectedResponse) {
+				t.Logf("found %d expected: %s", expectedMatches, line)
+				return true, nil
+			}
+		}
+		return false, nil
+	})
+	if err != nil {
+		t.Logf("Some errors were found: %v\nFor debugging purpose, please check the curl pod logs in the output.", err)
+		pod := &corev1.Pod{}
+		podName := types.NamespacedName{
+			Namespace: clientPod.Namespace,
+			Name:      clientPod.Name,
+		}
+		if err := kclient.Get(context.TODO(), podName, pod); err != nil {
+			t.Errorf("failed to get pod %s: %v", clientPod.Name, err)
+		}
+
+		logs, err := client.CoreV1().Pods(clientPod.Namespace).GetLogs(clientPod.Name, &corev1.PodLogOptions{
+			Container: "curl",
+			Follow:    false,
+		}).DoRaw(context.TODO())
+		if err != nil {
+			t.Errorf("failed to get logs from pod %s: %v", clientPod.Name, err)
+		}
+
+		t.Fatalf("failed to observe the expected output: %v\nclient pod spec: %#v\nclient pod logs:\n%s", err, pod, logs)
+	}
+}
+
+func TestSetIngressControllerResponseHeaders(t *testing.T) {
+	t.Parallel()
+	var headerNameXFrame string = "X-Frame-Options"
```

Comment:
Better yet:
```suggestion
	const (
		headerName  = "x-frame-options"
		headerValue = "DENY"
	)
```
And replace `"DENY"` with `headerValue` below.